File: Robot Mesh Studio/C++/V5 Competition Example.cpp
//LINK TO SOURCE: https://www.robotmesh.com/studio/5be40d5ec8f17a1f579d1787
// VEX V5 C++ Project with Competition Template
#include "vex.h"
using namespace vex;

//#region config_globals
vex::brain      Brain;
vex::motor      motor_right(vex::PORT15, vex::gearSetting::ratio18_1, false);
vex::motor      motor_left(vex::PORT16, vex::gearSetting::ratio18_1, true);
vex::drivetrain dt(motor_left, motor_right, 319.1764, 292.1, vex::distanceUnits::mm);
vex::controller con(vex::controllerType::primary);
//#endregion config_globals


// Creates a competition object that allows access to Competition methods.
vex::competition Competition;

void pre_auton() {
    // All activities that occur before competition start
    // Example: setting initial positions

}

void autonomous() {
    // Place autonomous code here
    dt.setVelocity(20,velocityUnits::pct);
    dt.driveFor(fwd,24,distanceUnits::in);
    dt.turnFor(turnType::left,180,deg);
    dt.driveFor(fwd,24,distanceUnits::in);
    dt.turnFor(turnType::left,180,deg);
}

void drivercontrol() {
    // Place drive control code here, inside the loop
    while (true) {
        // This is the main loop for the driver control.
        // Each time through the loop you should update motor
        // movements based on input from the controller.
        motor_left.spin(fwd,con.Axis3.position(pct),pct);
        motor_right.spin(fwd,con.Axis2.position(pct),pct);
    }
}

int main() {
    // Do not adjust the lines below

    // Set up (but don't start) callbacks for autonomous and driver control periods.
    Competition.autonomous(autonomous);
    Competition.drivercontrol(drivercontrol);

    // Run the pre-autonomous function.
    pre_auton();

    // Robot Mesh Studio runtime continues to run until all threads and
    // competition callbacks are finished.
    
    con.setDeadband(5,pct);
}

File: Robot Mesh Studio/Offical Vex examples/Python/V5 Competition Example.py
#LINK TO SOURCE: https://www.robotmesh.com/studio/5be40d5ec8f17a1f579d1784
import vex
import sys

#region config
brain       = vex.Brain();
motor_right = vex.Motor(vex.Ports.PORT15, vex.GearSetting.RATIO18_1, False)
motor_left  = vex.Motor(vex.Ports.PORT16, vex.GearSetting.RATIO18_1, True)
dt          = vex.Drivetrain(motor_left, motor_right, 319.1764, 292.1, vex.DistanceUnits.MM)
con         = vex.Controller(vex.ControllerType.PRIMARY)
#endregion config

competition = vex.Competition()

def driver():
  # Place drive control code here, inside the loop
  while True:
    # This is the main loop for the driver control.
    # Each time through the loop you should update motor
    # movements based on input from the controller.
    motor_left.spin(vex.DirectionType.FWD, (con.axis3.position(vex.PercentUnits.PCT)), vex.VelocityUnits.PCT)
    motor_right.spin(vex.DirectionType.FWD, (con.axis2.position(vex.PercentUnits.PCT)), vex.VelocityUnits.PCT)
competition.drivercontrol(driver)

def auto():
  # Place autonomous code here
  dt.drive_for(vex.DirectionType.FWD, 24, vex.DistanceUnits.IN)
  dt.turn_for(vex.TurnType.LEFT, 180, vex.RotationUnits.DEG)
  dt.drive_for(vex.DirectionType.FWD, 24, vex.DistanceUnits.IN)
  dt.turn_for(vex.TurnType.LEFT, 180, vex.RotationUnits.DEG)
competition.autonomous(auto)


# main thread
# All activities that occur before competition start
# Example: setting initial positions
con.set_deadband(5, vex.PercentUnits.PCT)
dt.set_velocity(20, vex.VelocityUnits.PCT)